### PR TITLE
Adding MFA scripts

### PR DESCRIPTION
**Pull request by iamanigeeit**
From https://github.com/espnet/espnet/issues/4521 : this commit adds scripts necessary to convert MFA to ESPnet formats.

- `mfa.sh` automates the entire conversion process
- `tts.sh` has been modified to pass arguments to `data.sh` for MFA
- `data.sh` has been edited to produce required text and durations files

To create .lab files for other datasets, define the relevant function in `mfa_format.py`.

### CODE REVIEW

File: egs2/TEMPLATE/asr1/scripts/utils/mfa.sh [hunk 1]
```diff
@@ -0,0 +1,68 @@
+set -e
+
+# If you are not using LJSpeech, be sure to define how your dataset is processed in mfa_format.py
+# Three
+
+#!/usr/bin/env bash
+# conda config --append channels conda-forge
+# conda install montreal-forced-aligner
```
Comment (kan-bayashi) [inline egs2/TEMPLATE/asr1/scripts/utils/mfa.sh:8]:
```suggestion
#!/usr/bin/env bash

# Generate MFA alignement

# You need to install the following tools to run this script:
# $ conda config --append channels conda-forge
# $ conda install montreal-forced-aligner

# If you are not using LJSpeech, be sure to define how your 
# dataset is processed in `scripts/utils/mfa_format.py`.

set -e
```

File: egs2/ljspeech/tts1/run_mfa.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+# Set bash to 'debug' mode, it will exit on :
+# -e 'error', -u 'undefined variable', -o ... 'error in pipeline', -x 'print commands',
```
[the remark this answers is not in the document]
Comment (kan-bayashi) [reply egs2/ljspeech/tts1/run_mfa.sh:3]: And if you make separated script, I think it is better to add all options in this scripts in order to train by just executing `./run_mfa.sh`.

File: egs2/TEMPLATE/asr1/scripts/utils/mfa.sh [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+#    --teacher_dumpdir data \
+#    --tts_stats_dir data/stats \
+#    --write_collected_feats true
+#./run_mfa.sh --stage 6 --stop_stage 6 \
+#    --train_config conf/tuning/train_fastspeech2.yaml \
+#    --teacher_dumpdir data \
+#    --tts_stats_dir data/stats
+
+
+
```
Comment (kan-bayashi) [inline egs2/TEMPLATE/asr1/scripts/utils/mfa.sh:68]:
```suggestion

echo "Successfully finished generating MFA alignments.

# NOTE(iamanigeeit): If you want to train FastSpeech2 with the alignments, 
# please check `egs2/ljspeech/tts1/run_mfa.sh` as an example.
```

File: egs2/TEMPLATE/asr1/scripts/utils/mfa_format.py [hunk 4]
```diff
@@ -0,0 +1,212 @@
+'''
+This script converts files to/from MFA format for use in ESPnet TTS.
+If you wish to add functions to create .lab files for MFA, add them like this:
+    `def make_labs_[dataset]:
+         ...`
+'''
```
Comment (kan-bayashi) [inline egs2/TEMPLATE/asr1/scripts/utils/mfa_format.py:6]:
```suggestion
#!/usr/bin/env python3

"""Convert files to/from MFA format for use in ESPnet TTS.

If you wish to add functions to create .lab files for MFA, add them like this:

def make_labs_[dataset]:
     ...
"""
```

**Review by kan-bayashi — CHANGES_REQUESTED** [review]
Sorry for the late reviewing and thank you for adding cool features!
Could you reflect my comments and fix the CI errors?

**Comment by iamanigeeit** [comment]
@kan-bayashi Sorry for the delay because of ICASSP submission. I will pull the latest version of ESPnet and fix my PR this week!

**Comment by iamanigeeit** [comment]
Hello @kan-bayashi 

I have made the requested changes, but i can't submit the PR, because i have modified other files for my own work on `iamanigeeit:master`.

I tried to fork `espnet` again, but i am only allowed to fork once. When i create a new branch `iamanigeeit:espnet/mfa` and try to create a new PR from there, i get

> Can't create a new pull request: Although you appear to have the correct authorization credentials, the `espnet` organization has enabled OAuth App access restrictions, meaning that data access to third-parties is limited.

What should i do?

**Comment by kan-bayashi** [comment]
Thank you for your modification, @iamanigeeit.

> Can't create a new pull request: Although you appear to have the correct authorization credentials, the espnet organization has enabled OAuth App access restrictions, meaning that data access to third-parties is limited.

I've never met this kind of error. Let me confirm your procedure.

1. make new branch and perform cherry-pick in your local
2. push your branch to your fork `iamanigeeit/espnet`.
3. make PR from github HP

Is it correct?

**Comment by iamanigeeit** [comment]
Thanks @kan-bayashi 

Creating a new branch and making the PR from Github instead of Pycharm seems to work. The new commit is at https://github.com/espnet/espnet/pull/4801

**Pull request closed without merging**